test/jdk/java/util/Calendar/CalendarLeapYearAddTest.java

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright (c) 2024, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+ * @test
+ * @bug 8331646
+ * @summary confirm that Calendar.add() works correctly with leap year calculations
+ * @run junit CalendarLeapYearAddTest
+ */
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import java.util.Calendar;
+import java.util.Locale;
+import java.util.stream.Stream;
+
+import static java.util.Calendar.*;
+import static java.util.Calendar.MONTH;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class CalendarLeapYearAddTest {
+
+    /**
+     * 8331646 Calendar add for leap year
+     */
+    @ParameterizedTest
+    @MethodSource("calendarAddSource")
+    public void testAddLeapYear(String testName, int calendarDate, int calendarMonth, int calendarYear,
+                                int value, int calendarField, int expectedDate, int expectedMonth,
+                                int expectedYear) {
+        Calendar calendar = Calendar.getInstance();
```

Comment:
Instead of using a factory on `Calendar`, this should explicitly construct a `GregorianCalendar` instance which guarantees the leap-year behavior.